Mage/src/main/java/mage/abilities/effects/PreventDamageAndRemoveCountersEffect.java

```diff
@@ -38,19 +51,41 @@ public PreventDamageAndRemoveCountersEffect copy() {
 
     @Override
     public boolean replaceEvent(GameEvent event, Ability source, Game game) {
-        int damage = event.getAmount();
+        int damageAmount = event.getAmount();
         preventDamageAction(event, source, game);
         Permanent permanent = game.getPermanent(source.getSourceId());
-        if (permanent == null) {
+        PreventDamageAndRemoveCountersWatcher watcher = game.getState().getWatcher(PreventDamageAndRemoveCountersWatcher.class);
+        if (permanent == null || watcher == null || damageAmount <= 0) {
             return false;
         }
-        if (!thatMany) {
-            damage = 1;
+        MageObjectReference mor = new MageObjectReference(source.getId(), source.getSourceObjectZoneChangeCounter(), game);
+        int beforeCount = permanent.getCounters(game).getCount(CounterType.P1P1);
+        if (thatMany) {
+            // Remove them.
+            permanent.removeCounters(CounterType.P1P1.createInstance(damageAmount), source, game);
+        } else if (!watcher.hadMORCounterRemovedThisBatch(mor)) {
```

Comment:
Hmmm. Does `whileHasCounter` in the `applies` method work properly for `thatMany` in the case of simultaneous damage? Because it should all be prevented, even if the damage from one source would remove all the counters. Please write unit test to confirm.

(If it doesn't work, then probably need to check the watcher in all cases, and move the `whileHasCounter` check from `applies` into the replacement, checking both it and the watcher before the prevent damage action. But I'm hoping it works as actions not processed in between.)

Reply:
On point, that was bugged.
Have fixed & tested it. If in the watcher's Set, should applies even with no counters, as the current batch already removed a counter, meaning there was one at the start.